osi_logicallane.proto

```diff
@@ -0,0 +1,502 @@
+// - The LogicalLaneBoundary has ST coordinates, and is thus a separate type
+//   from LaneBoundary.
+//   Advantages of this decision:
+//   - Calculations like getting the lane width at a position are easy, since
+//     one can just get the boundary points of the left and right boundary at
+//     the desired S position (via linear interpolation), and calculate the
+//     width from the two points. Also getting something like the distance to
+//     the lane border is very easy.
+//   - No centerline of the lane is necessary, since this can very easily be
+//     generated from the boundaries.
+//   Disadvantages of this decision:
+//   - Lane boundaries cannot be shared with physical lanes. This results in
+//     more data needed. This can mostly be mitigated by only transmitting the
+//     lane boundaries during initialization (e.g. via the OSMP GroundTruthInit
+//     message).
+// - The LogicalLaneBoundary contains all data directly which an agent model is
+//   likely to need. It does not include information normally only used by
+//   sensor models (e.g. the exact length of the color markings on the road).
+//   This information can be gotten from the physical lane referenced in the
+//   LogicalLane, if needed.
+//
+// TODO To Discuss: It would be entirely possible to add ST coordinates also to
+// LaneBoundary. Then no separate type would be needed. However, it is unclear if
+// the additional overhead is acceptable for sensor models.
+//
+message LogicalLaneBoundary
+{
+    // The ID of the lane boundary.
+    //
+    // \rules
+    // is_globally_unique
+    // \endrules
+    //
+    optional Identifier id = 1;
+
+    // points making up the lane boundary.
+    // The boundary must be defined in the same direction as the reference
+    // line. So S positions should increase along the line. Note that S
+    // positions will not always increase strictly monotonically.
+    // Example:
+    //                |---------\
+    //                |          \
+    //  /---boundary--|           \------------
+    // /
+    // ---------- reference line --------------
+    // At the place where the boundary suddenly increases, two points will have
+    // the same S coordinate.
+    //
+    // If the boundary approximates a curve (e.g. a cubic function in
+    // OpenDRIVE), the points must be chosen in a way that the lateral distance
+    // to the ideal line does not exceed 5cm. As shown in the following image:
+    //
+    // \image html line_approximation_error.svg "Approximation error"
+    // Approximation error green line.
+    //
+    // The Z error (difference in Z height between boundary_line and the "real"
+    // line) must also not exceed 5cm.
+    //
+    // Note: if two lanes have different Z heights (e.g. a driving lane is
+    // beside a sidewalk, where the sidewalk is 10cm higher than the road),
+    // then these lanes cannot share a boundary, since their boundaries have
+    // different Z heights.
```

Comment:
That's a good comment, which we should also add to the physical boundaries IMO.

Reply:
Sounds reasonable. I won't do it for this pull request (I don't want to make it larger or more contentious than necessary). Do you want to create a pull request for this?

Reply:
Yes makes sense, I'll do and close this comment